stablehlo/reference/Tensor.h

```diff
@@ -67,38 +68,45 @@ class Tensor {
   Tensor();
   explicit Tensor(ShapedType type);
   explicit Tensor(DenseElementsAttr attr);
-  Tensor(const Tensor &other) = default;
+  Tensor(const Tensor &other);
   /// @}
 
   /// Assignment operator.
-  Tensor &operator=(const Tensor &other) = default;
+  Tensor &operator=(const Tensor &other);
 
   /// Returns type of the Tensor object.
   ShapedType getType() const;
 
   /// Returns the number of elements.
   int64_t getNumElements() const;
 
-  /// Provides read access, via a linearized element index, into the
-  /// underlying storage.
-  Element get(int64_t index) const;
+  /// Provides read access to the tensor element indexed at 'indices'.
+  Element get(llvm::ArrayRef<int64_t> indices) const;
 
-  /// Provides write access, via a linearized element index, into the
-  /// underlying storage.
+  /// Provides write access to the tensor element indexed at 'indices'.
   ///
-  /// \param index The index to write to.
+  /// \param indices The multi-dimensional index to write to.
   /// \param element The Element object \a element is used to update the
-  /// underlying storage pointed to by \a index.
-  void set(int64_t index, Element element);
+  /// underlying storage pointed to by \a indices.
+  void set(llvm::ArrayRef<int64_t> indices, const Element &element);
 
   /// Print utilities for Tensor objects.
   void print(raw_ostream &os) const;
 
   /// Print utilities for Tensor objects.
   void dump() const;
 
+  /// Iterator utils.
+  IndexSpaceIterator indexBegin() const;
+  IndexSpaceIterator indexEnd() const;
```

Comment:
LLVM's code style has a carveout for iterator methods - https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly - so these should be called `index_begin` and `index_end`.